### PR TITLE
Use original name when checking allowlist for anonymous enum variants

### PR DESCRIPTION
**Pull request by jethrogb**
Prior to this PR, this header

```c
enum {
    MyVal = 0,
};
```

in combination with `allowlist_var("^MyVal$")` and an implementation of `ParseCallbacks::enum_variant_name` would generate no code. This is because the renamed variant name is checked against the variable allowlist. All other allowlist checks are against the unrenamed name. This PR makes that consistent. This is a breaking change but I think the old behavior is clearly wrong.

Additionally, I've added a way to test ParseCallbacks in CI.

### CODE REVIEW

**Review by emilio — COMMENTED** [review]
So this looks sensible, but regarding testing: The way we've been testing these has been using the `bindgen-integration` crate. It seems it'd be easy to add a test for this there rather than adding new testing infrastructure, wdyt?

**Comment by jethrogb** [comment]
> regarding testing

I see, I didn't realize that existed. Looking at it briefly, the bindgen-integration is quite monolithic and it doesn't check the exact output. I think the expectation testing framework is much better, so I'd prefer to do it this way. But if you feel strongly about it I can move the tests to bindgen-integration.

**Comment by emilio** [comment]
> > regarding testing
> 
> I see, I didn't realize that existed. Looking at it briefly, the bindgen-integration is quite monolithic and it doesn't check the exact output. I think the expectation testing framework is much better, so I'd prefer to do it this way. But if you feel strongly about it I can move the tests to bindgen-integration.

Ok, I'm fine with that, it's not too complicated anyways. Thanks!

File: tests/tests.rs [hunk 1]
```diff
@@ -290,6 +314,7 @@ fn create_bindgen_builder(header: &PathBuf) -> Result<Builder, Error> {
 
     // Scoop up bindgen-flags from test header
     let mut flags = Vec::with_capacity(2);
+    let mut parse_callbacks = vec![];
```
Comment (emilio) [inline tests/tests.rs:317]: Hmm, is there any need for handling multiple of these? The last one will just win right? Can this just use an `Option`?

Comment (emilio) [inline tests/tests.rs:317]: Hmm, is there any need for handling multiple of these? The last one will just win right? Can this just use an `Option`?

**Review by emilio — COMMENTED** [review]
Just one question, but looks fine otherwise.

File: tests/tests.rs (hunk 1 above)
Comment (jethrogb) [reply tests/tests.rs:317]: Oh indeed. Somehow I though all would get called but obviously that doesn't really make sense.

**Review by emilio — APPROVED** [review]

**Pull request merged**